src/client.ts

```diff
@@ -5361,6 +5364,48 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
             ?? timeline;
     }
 
+    /**
+     * Get an EventTimeline for the latest events in the room. This will just
+     * call `/messages` to get the latest message in the room, then use
+     * `client.getEventTimeline(...)` to construct a new timeline from it.
+     *
+     * @param {EventTimelineSet} timelineSet  The timelineSet to look for the
+     * event in
+     * @param {string} eventId  The ID of the event to look for
+     *
+     * @return {Promise} Resolves:
+     *    {@link module:models/event-timeline~EventTimeline} including the given
+     *    event
+     */
+    public async getLatestTimeline(timelineSet: EventTimelineSet): Promise<EventTimeline> {
```

Comment:
This feels like a racey concept, whilst you're waiting for the response from /context new messages may have come in which now won't be on the tip of this timeline.

Reply:
We should continue this conversation in the other thread, https://github.com/matrix-org/matrix-js-sdk/pull/2299#discussion_r852891150

If you have certain scenarios you're worried about, I can add a test for it.